crates/red_knot_python_semantic/resources/mdtest/scopes/global.md

```diff
@@ -0,0 +1,200 @@
+# `global` references
+
+## Implicit global in function
+
+A name reference to a never-defined symbol in a function is implicitly a global lookup.
+
+```py
+x = 1
+
+def f():
+    reveal_type(x)  # revealed: Unknown | Literal[1]
+```
+
+## Explicit global in function
+
+```py
+x = 1
+
+def f():
+    global x
+    reveal_type(x)  # revealed: Unknown | Literal[1]
+```
+
+## Unassignable type in function
+
+```py
+x: int = 1
+
+def f():
+    y: int = 1
+    # error: [invalid-assignment] "Object of type `Literal[""]` is not assignable to `int`"
+    y = ""
+
+    global x
+    # @Todo(error: [invalid-assignment] "Object of type `Literal[""]` is not assignable to `int`")
+    x = ""
+```
+
+## Nested intervening scope
+
+TODO this should give the outer module type of `Unknown | Literal[1]`, not `Literal[""]`
+
+```py
+x = 1
+
+def outer():
+    x = ""
+
+    def inner():
+        global x
+        reveal_type(x)  # revealed: Unknown | Literal[""]
+```
+
+## Narrowing
+
+An assignment following a `global` statement should narrow the type in the local scope after the
+assignment. The revealed type of `Literal[1]` here is in line with [pyright], while [mypy] reports
+`builtins.int`.
+
+```py
+x: int | None
+
+def f():
+    global x
+    x = 1
+    reveal_type(x)  # revealed: Literal[1]
+```
+
+This related case is adapted from a `mypy` [test][t] with the comment:
+
+> This is unsafe, but we don't generate an error, for convenience. Besides, this is probably a very
+> rare case.
+
+```py
+g: str | None
+
+def f():
+    global g
+    g = "x"
+    def nested() -> str:
+        return g
+```
+
+## `nonlocal` and `global`
+
+A binding cannot be both `nonlocal` and `global`. This should emit a semantic syntax error. CPython
+marks the `nonlocal` line, while `mypy`, `pyright`, and `ruff` (`PLE0115`) mark the `global` line.
+
+```py
+x = 1
+
+def f():
+    x = 1
+    def g() -> None:
+        nonlocal x
+        global x  # TODO error: [invalid-syntax] "name 'x' is nonlocal and global"
+        x = None
+```
+
+## Global declaration after `global` statement
+
+This is also adapted from a `mypy` [test].
```

Comment:
This is a pretty basic unremarkable test for how `global` should work -- it's fine that you happened to get it from mypy, but I don't think it's useful to preserve that information here.

```suggestion
```